src/effects/backends/builtin/compressoreffect.cpp

```diff
@@ -0,0 +1,215 @@
+#include "effects/backends/builtin/compressoreffect.h"
+
+// static
+QString CompressorEffect::getId() {
+    return "org.mixxx.effects.compressor";
+}
+
+//static
+EffectManifestPointer CompressorEffect::getManifest() {
+    EffectManifestPointer pManifest(new EffectManifest);
+    pManifest->setId(getId());
+    pManifest->setName(QObject::tr("Compressor"));
+    pManifest->setShortName(QObject::tr("Compressor"));
+    pManifest->setAuthor("The Mixxx Team");
+    pManifest->setVersion("1.0");
+    pManifest->setDescription("A single-band compressor effect");
+    pManifest->setEffectRampsFromDry(true);
+    pManifest->setMetaknobDefault(0.0);
+
+    EffectManifestParameterPointer clipping = pManifest->addParameter();
+    clipping->setId("clipping");
+    clipping->setName(QObject::tr("Clipping"));
+    clipping->setShortName(QObject::tr("Clipping"));
+    clipping->setDescription(QObject::tr("Hard limiter to 0 db"));
```

Comment:
Is that ever useful? A hard clipper will introduce additional frequencies.  Did you consider a tanh clipper, that is close to analogue circuit?

Reply:
Thank you, it's a good point. Actually I added it just to avoid sound going to "red" level, for example if there are some very fast peaks, which compressor didn't catch because of the slow attack time. And no, I haven't considered any other clippers yet.

Despite the fact that it's a very simple solution, I believe it's useful. We could introduce more advanced solution later, for example in this feature request: #6513.